src/pyaro_readers/nilupmfebas/_lowlevel_helpers.py

```diff
@@ -0,0 +1,715 @@
+    if not os.path.exists(d):
+        os.mkdir(d)
+    return d
+
+
+def check_dirs_exist(*dirs, **add_dirs):
+    for d in dirs:
+        if not os.path.exists(d):
+            print(f"Creating dir: {d}")
+            os.mkdir(d)
+    for k, d in add_dirs.items():
+        if not os.path.exists(d):
+            os.mkdir(d)
+            print(f"Creating dir: {d} ({k})")
+
+
+def list_to_shortstr(lst, indent=0):
+    """Custom function to convert a list into a short string representation"""
+
+    def _short_lst_fmt(lin):
+        lout = []
+        for val in lin:
+            try:
+                with ignore_warnings(
+                    RuntimeWarning,
+                    "divide by zero encountered in log10",
+                    "overflow encountered in long_scalars",
+                ):
+                    ndigits = (
+                        -1 * np.floor(np.log10(abs(np.asarray(val)))).astype(int) + 2
+                    )
+                lout.append(f"{val:.{ndigits}f}")
+            except Exception:
+                lout.append(val)
+        return lout
+
+    name_str = f"{type(lst).__name__} ({len(lst)} items): "
+    indentstr = indent * " "
+    if len(lst) == 0:
+        return f"{indentstr}{name_str}[]"
+    elif len(lst) < 6:
+        lfmt = _short_lst_fmt(lst)
+        return f"{indentstr}{name_str}{lfmt}"
+    else:  # first 2 and last 2 items
+        lfmt = _short_lst_fmt([lst[0], lst[1], lst[-2], lst[-1]])
+        s = f"{indentstr}{name_str}[{lfmt[0]}, {lfmt[1]}, ..., {lfmt[2]}, {lfmt[3]}]"
+
+    return s
+
+
+def sort_dict_by_name(d, pref_list: list = None) -> dict:
+    """Sort entries of input dictionary by their names and return ordered
+
+    Parameters
+    ----------
+    d : dict
+        input dictionary
+    pref_list : list, optional
+        preferred order of items (may be subset of keys in input dict)
+
+    Returns
+    -------
+    dict
+        sorted and ordered dictionary
+    """
+    if pref_list is None:
+        pref_list = []
+    s = {}
+    sorted_keys = sorted(d)
+    for k in pref_list:
+        if k in d:
+            s[k] = d[k]
+    for k in sorted_keys:
+        if not k in pref_list:
+            s[k] = d[k]
+    return s
+
+
+def dict_to_str(dictionary, indent=0, ignore_null=False):
+    """Custom function to convert dictionary into string (e.g. for print)
+
+    Parameters
+    ----------
+    dictionary : dict
+        the dictionary
+    indent : int
+        indent of dictionary content
+    ignore_null : bool
+        if True, None entries in dictionary are ignored
+
+    Returns
+    -------
+    str
+        the modified input string
+
+    """
+    if len(dictionary) == 0:
+        return "{}"
+    elif len(dictionary) == 1:
+        pre = ind = offs = ""
+    else:
+        pre = "\n"
+        ind = indent * " "
+        offs = " "
+    s = "{"
+
+    for key, val in dictionary.items():
+        if ignore_null and val is None:
+            continue
```

Comment:
Is all of this code actually needed by pyaro-readers?

Reply:
I have removed all unused code that came from pyaerocom